website/docs/r/redhatopenshift_cluster.html.markdown

```diff
@@ -0,0 +1,248 @@
+}
+
+resource "azurerm_subnet" "worker_subnet" {
+  name                 = "worker-subnet"
+  resource_group_name  = azurerm_resource_group.example.name
+  virtual_network_name = azurerm_virtual_network.example.name
+  address_prefixes     = ["10.0.0.0/23"]
+  service_endpoints    = ["Microsoft.Storage", "Microsoft.ContainerRegistry"]
+}
+
+resource "azurerm_redhat_openshift_cluster" "example" {
+  name                = "example-redhatopenshift1"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+
+  cluster_profile {
+    domain = "foo.example.com"
+  }
+
+  service_principal {
+    client_id     = "00000000-0000-0000-0000-000000000000"
+    client_secret = "00000000000000000000000000000000"
+  }
+
+  main_profile {
+    vm_size   = "Standard_D8s_v3"
+    subnet_id = azurerm_subnet.main_subnet.id
+  }
+
+  worker_profile {
+    vm_size      = "Standard_D4s_v3"
+    disk_size_gb = 128
+    subnet_id    = azurerm_subnet.worker_subnet.id
+    node_count   = 3
+  }
+  
+  api_server_profile {
+    visibility = "Public"
+  }
+
+  ingress_profile {
+    visibility = "Public"
+  }
+
+  tags = {
+    Environment = "Production"
+  }
+  
+  depends_on = ["azurerm_role_assignment.redhatopenshift"]
+}
+
+output "console_url" {
+  value = azurerm_redhatopenshift_cluster.example.console_url
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Azure Red Hat Openshift Cluster to create. Changing this forces a new resource to be created.
+
+* `location` - (Required) The location where the Azure Red Hat Openshift Cluster should be created. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) Specifies the Resource Group where the Azure Red Hat Openshift Cluster should exist. Changing this forces a new resource to be created.
+
+* `service_principal` - (Required) A `service_principal` block as defined below.
+
+* `main_profile` - (Required) A `main_profile` block as defined below.
+
+* `worker_profile` - (Required) A `worker_profile` block as defined below.
+
+* `cluster_profile` - (Required) A `cluster_profile` block as defined below.
+
+* `api_server_profile` - (Required) An `api_server_profile` block as defined below.
+
+* `ingress_profile` - (Required) An `ingress_profile` block as defined below.
+
+* `network_profile` - (Optional) A `network_profile` block as defined below.
+
+* `tags` - (Optional) A mapping of tags to assign to the resource.
+
+---
+
+A `service_principal` block supports the following:
+
+* `client_id` - (Required) The Client ID for the Service Principal. Changing this forces a new resource to be created.
+
+* `client_secret` - (Required) The Client Secret for the Service Principal. Changing this forces a new resource to be created.
+
+~> **Note:** Currently a service principal cannot be associated with more than one ARO clusters on the Azure subscription.
```

Comment:
This service principal is also given Contributor access to ARO, by default, when creating via azure portal.